service/cloudfront/sign/sign_cookie.go

```diff
@@ -0,0 +1,141 @@
+// 	p := &sign.Policy{
+// 		Statements: []sign.Statement{
+// 			{
+//				Resource: RawCloudFrontURL, //read the provided documentation on how to set this correctly, you'll probably want to use wildcards
+// 				Condition: sign.Condition{
+// 					// Optional IP source address range
+// 					IPAddress: &sign.IPAddress{SourceIP: "192.0.2.0/24"},
+// 					// Optional date URL is not valid until
+// 					DateGreaterThan: &sign.AWSEpochTime{time.Now().Add(30 * time.Minute)},
+// 					// Required date the URL will expire after
+// 					DateLessThan: &sign.AWSEpochTime{time.Now().Add(1 * time.Hour)},
+// 				},
+// 			},
+// 		},
+// 	}
+
+// 	//load your private key and convert it to type rsa.PrivateKey
+// 	privKey, err := sign.LoadPEMPrivKeyFile("privatekey.pem")
+// 	if err != nil {
+// 		fmt.Println("error", err)
+// 	}
+
+// 	//key ID that represents the key pair associated with the private key
+// 	keyID := "WOIERULSDLKEWRIU"
+
+// 	//set credentials to the cookiesigner
+// 	signer := sign.NewCookieSigner(keyID, privKey)
+
+// 	//creates 3 signed cookies, provide an optional Cookie Options struct to specify other options
+// 	o := &sign.CookieOptions{
+// 		Path:   "/",
+// 		Domain: ".cNameAssociatedWithMyDistribution.com",
+// 		Secure: true, //make sure your app/site can handle https payloads, otherwise set this to false
+// 	}
+// 	//http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-setting-signed-cookie-custom-policy.html#private-content-custom-policy-statement-signed-cookies-examples
+//  //http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/CNAMEs.html
+// 	//avoid adding an Expire or MaxAge. See provided AWS Documentation for more info
+
+// 	policy, signature, key, err := signer.SignCookies(p)
+// 	if err != nil {
+// 		fmt.Println("error", err)
+// 	}
+
+// 	http.SetCookie(w, policy)
+// 	http.SetCookie(w, signature)
+// 	http.SetCookie(w, key)
+
+// }
+
+// NewCookieSigner constructs and returns a new CookieSigner to be used to for signing
+// Amazon CloudFront URL resources with.
+func NewCookieSigner(keyID string, privKey *rsa.PrivateKey) *CookieSigner {
+	return &CookieSigner{
+		keyID:   keyID,
+		privKey: privKey,
+	}
+}
+
+//SignCookies CookieOptions are optional, if unused, just pass in nil
+func (c CookieSigner) SignCookies(p *Policy, o *CookieOptions) (cPolicy, cSignature, cKey *http.Cookie, err error) {
+	b64Sig, b64Policy, err := p.Sign(c.privKey)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	cPolicy, cSignature, cKey = c.createCookies(b64Policy, b64Sig, o)
+	return cPolicy, cSignature, cKey, nil
+
+}
+
+//prepares the cookies to be attached to the header. An (optional) options struct is provided
+//in case people don't want to manually edit their cookies
+func (c CookieSigner) createCookies(policy, signature []byte, o *CookieOptions) (cPolicy, cSignature, cKey *http.Cookie) {
```

Comment:
What do you think about using a `[]*http.Cookie` instead of explicit values? All cookies returned are required to be added, so a list that can be iterated over might be easier to use.